backend/internal/service/handler/users/user.go

```diff
@@ -125,6 +125,44 @@ func (h *Handler) GetUserProfile(c *fiber.Ctx) error {
 	return c.Status(fiber.StatusOK).JSON(profile)
 }
 
+
+func (h *Handler) UpdateUserProfilePicture(c *fiber.Ctx) error {
+	id := c.Params("id")
+
+	exists, err := h.userRepository.UserExists(c.Context(), id)
```

Comment:
### Two-Step Query Pattern

1. check query
2. act query

IMO this pattern is a code smell. your main query will fail anyways if the user doesn't exist. 

### Reasoning

1. **Increased latency**: each query incurs network and/or database overhead, which adds up if every request requires two separate queries.
2. **Concurrency problems**: between the "check" and act" steps, the state of your entity, in this case user, could change. for example, if another request modifies or deletes the user. this leads to potential **race conditions**
3. **Database Load**: double queries can put unnecessary load on the database, especially if the same information could be derived in a single query.

### Alternatives

1. **Atomic Operations**: where possible, use a single, atomic query that combines the check and the act. for example, an `UPSERT` (insert if not exists, update if it does) operation, or a conditional query
2. **Error Handling for Non-Existence**: If the second query fails because the user does not exist, handle this as an expected error. This avoids the need for a preliminary existence check and keeps the operation concise.
3. **Optimistic or Pessimistic Locking**: for cases where race conditions are a concern, applying locking mechanisms can ensure operational integrity.

### When a Two-Step Check Might Be Justified

1. When the "check" query is caching the result, so the "act" query is conditionally executed based on that.
2. For high-read, low-write data where the probability of conflict is low and the cost of combining the queries outweighs potential issues.